### DISCUSSION TITLE
Source of ballistic data?

### DISCUSSION THREAD

**Discussion by kevin-dolan** [tarkov-ballistics/website#2] [Q&A]
### tl;dr How/where are ballistics data and algorithms being collected for this project?
Hello,
I'm very interested in how this project gets the bullet/armor stats and ballistic calculations that are used in-game. The wiki (on fandom.com) has [a long page](https://escapefromtarkov.fandom.com/wiki/Ballistics) on how in-game ballistics work, but **does not** list any sources for data not listed in-game such as penetration values, fragmentation chance, armor durability damage % or the exact formulae for penetration chance against damaged armor at a given durability level. The ballistics page references an [article](https://tarkovaftermidnight.wordpress.com/) by streamer NoFoodAfterMidnight, but the article was last updated on 5/29/2020, over a year and a half ago and long before the 12.12 ballistics changes. 
It seems as though this project uses the most up to date data and algorithms to calculate important ballistic stats like % to pen, damage to armor durability, and damage falloff over distance, but I deeply desire to see the _exact_ algorithms for certain things like damage to armor durability. Veritas' [BattleBuddy app](https://play.google.com/store/apps/details?id=com.veritas.battlebuddy&hl=en_US&gl=US) does actually allow you to simulate damage to armor durability for a given ammo type, but I'm afraid that it's out of date like the wiki. 
All that being said, I'd like to either know the process by which this project got its stats and algorithms so I can collect it myself, or I would like to just get all of that raw data dumped in a zip file or something. This desire for hyper-specific ballistic information came from looking at 12g ammo and desperately wanting viable ammo types besides Flechette and AP20. :(

**Comment by morphy2k** [comment]
I am also the developer of the Battle Buddy Android app, so I can assure you that the penetration calculator is still up-to-date. The health calculator needs an update, because the damage overflow is no longer distributed correctly, there will be an update soon.

We calculate all ballistic data with our self-written software using the original item properties. At a later date, we will make our software available as OSS.
At the current time, the ballistics system is a black box.

If you are missing something on our website, feel free to let us know.

**Comment by kevin-dolan** [comment]
With regards to something missing from the website: I would like to know how more specific numbers about average number of shots to 85 damage. On your site, there is a page for "Average Shots to 85 Damage By Class", where at the bottom is a note: "All values are average (Mean) values and were calculated at a distance of 30m. When calculating these values, all factors such as penetration chance and fragmentation were simulated, except for ricochets. Consider this chart as experimental." Damage reduction via armor is probably already taken into account, but I just want to double check.
Furthermore, I'd like to know more specifically what the _maximum_ number of shots to 85 damage with 95% and/or 99% confidence. I noticed that if you mouse-over the listed numbers, it will display a min/max value, but no confidence level is specified. Tarkov's armor penetration system works in such a way that armor of a lower class/damaged armor then an ammo's pen value still has a small but significant chance to block the shot. 
I also have one extremely petty comment about the presentation of the min/max numbers: I believe they should be whole integers rather than decimals. In Tarkov, a bullet hitting is a discrete event that can't partially happen (ignoring ricochets lol), so if the max penetrating bullets it takes to get 85 damage is calculated as 3.2, then it would be appropriate to display the max as 4 bullets. Granted, that would be inappropriate for the mean shots-to-85, which represents something more continuous. Also granted, "Consider this chart as experimental."

On a different note, when you say the ballistics system is a black box, do you mean that _your_ code is private or do you mean EFT's code for the ballistics system is completely inaccessible? If the later, from what source do you base your simulation on? Has BSG ever posted an exact breakdown of the algorithms used in game in a forum post or elsewhere? Do the item properties happen to contain information about the algorithms that use them? I ask mainly in case I want to calculate a hyper-specific scenario that is not covered by your site's or the BattleBuddy app's information. I want to do that sort of thing because I can't hit kill any PMCs and want to cope by calculating overly complicated scenarios where I did enough damage to them.
